test/splitter.js

```diff
@@ -0,0 +1,197 @@
+    it("Transaction reverts if the sender's address is also a recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(sender, recipientTwo, {from: sender, value: 4}),
+            "The sender cannot also be a recipient"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("SplitDeposit is pausable", async () => {
+
+        await splitter.pause({from: contractOwner});
+
+        await truffleAssert.reverts(
+            splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 4}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("SplitDeposit is unpausable", async () => {
+
+        await splitter.pause({from: contractOwner});
+        await splitter.unpause({from: contractOwner});
+
+        const txObj = await splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 5});
+
+        truffleAssert.eventEmitted(txObj, 'Deposit');
+    })
+
+
+    it('Sender deposits 5 wei and the recipient can successfully withdraw 2 wei', async () => {
+
+        const withDrawAmount = toBN(2);
+        
+        // Deposit into the contract
+        await splitter.splitDeposit(recipientOne, recipientTwo, {from: sender, value: toBN(5)});
+
+        // Take a snapshot of the second recipient's ETH balance
+        const initRecipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+
+        // Recipient withdraws
+        const txObj = await splitter.withdraw(withDrawAmount, {from: recipientTwo});
+        const cost = await getGasCost(txObj);
+
+        // Get the recipient's new ETH and contract balances
+        const recipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+        const recipientTwoOwed = toBN(await splitter.balances(recipientTwo));
+
+        // Calculate the expected new ETH and contract balances
+        const expectedRecipientTwoEthBalance = initRecipientTwoEthBalance.sub(cost).add(withDrawAmount).toString(10);
+
+        assert.strictEqual(recipientTwoEthBalance.toString(10), expectedRecipientTwoEthBalance);
+        assert.strictEqual(recipientTwoOwed.toString(10), "0");
+
+        truffleAssert.eventEmitted(txObj, "WithDraw", (ev) => {
+            return  ev.withdrawer === recipientTwo &&
+                    ev.amount.toString(10) === "2";
+        }, 'TestEvent should be emitted with correct parameters');
+
+    });
+
+    it("Second recipient attempts to withdraw 3 wei when there is only 3 wei", async () => {
+
+        await splitter.splitDeposit(recipientOne, recipientTwo, {from: sender, value: toBN(5)});
+
+        await truffleAssert.reverts(
+            splitter.withdraw(toBN(3), {from: recipientTwo}),
+            "There are insufficient funds"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Second recipient attempts to withdraw zero", async () => {
+
+        await truffleAssert.reverts(
+            splitter.withdraw(toBN(0), {from: recipientTwo}),
```

Comment:
This test would be more meaningful if the contract had a non-zero balance.

Reply:
Understood.